packages/kbn-plugin-system/src/plugin_spec.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  PluginsType,
+  PluginName,
+  PluginConfigPath,
+  KibanaClassPluginStatic,
+  KibanaPlugin,
+} from './plugin';
+
+const noop = () => {};
+
+export class PluginSpec<C, D extends PluginsType, E> {
+  readonly name: PluginName;
+  readonly dependencies: PluginName[];
+  readonly configPath?: PluginConfigPath;
+
+  private readonly _pluginClass: KibanaClassPluginStatic<C, D, E>;
+  private _pluginInstance?: KibanaPlugin<C, D, E>;
+  private _exposedValues?: E;
+
+  constructor(
+    name: PluginName,
+    config: {
+      configPath?: PluginConfigPath;
```

Comment:
That said, since we have the plugin name in the `kibanaPluginApiFactory` I wonder what this intention behind this property is.

Reply:
++ on `metadata` or similar. I'll explore some changes.

Reply:
You can see how it was used in the `new-platform` branch here: https://github.com/elastic/kibana/blob/new-platform/platform/src/server/plugins/kibana_plugin_values.ts#L17. I'm trying to not unnecessarily pass the full config service into plugins, and rely on the `configPath` defined like this: https://github.com/elastic/kibana/blob/new-platform/example_plugins/foo/src/index.ts#L7.

It might not necessarily make sense for the UI though.